storage/memoized_iterator.go

```diff
@@ -132,13 +126,17 @@ func (b *MemoizedSeriesIterator) At() (int64, float64) {
 	return b.it.At()
 }
 
-// AtHistogram returns the current histogram element of the iterator.
+// AtHistogram is not supported by this iterator.
 func (b *MemoizedSeriesIterator) AtHistogram() (int64, *histogram.Histogram) {
-	return b.it.AtHistogram()
+	panic("AtHistogram is not supported by this iterator.")
```

Comment:
If `MemoizedSeriesIterator` is used only in PromQL, maybe we could introduce a dedicated interface which does not expose `AtHistogram` at all.

It also feels strange that `Next` can return `ValueTypeHistogram` but calling `AtHistogram` would panic.

Reply:
Hm, that's a good point--I'll look into it